files.go

```diff
@@ -25,59 +27,66 @@ var FilesSchema = sql.Schema{
 	{Name: "blob_size", Type: sql.Int64, Source: "files"},
 }
 
-func newFilesTable() Indexable {
+func newFilesTable() *filesTable {
 	return new(filesTable)
 }
 
-var _ sql.PushdownProjectionAndFiltersTable = (*filesTable)(nil)
+var _ Table = (*filesTable)(nil)
 var _ Squashable = (*filesTable)(nil)
 
-func (filesTable) isGitbaseTable()      {}
-func (filesTable) isSquashable()        {}
-func (filesTable) Resolved() bool       { return true }
-func (filesTable) Name() string         { return FilesTableName }
-func (filesTable) Schema() sql.Schema   { return FilesSchema }
-func (filesTable) Children() []sql.Node { return nil }
+func (filesTable) isGitbaseTable()    {}
+func (filesTable) isSquashable()      {}
+func (filesTable) Name() string       { return FilesTableName }
+func (filesTable) Schema() sql.Schema { return FilesSchema }
 
-func (t *filesTable) TransformExpressionsUp(f sql.TransformExprFunc) (sql.Node, error) {
-	return t, nil
+func (r *filesTable) WithFilters(filters []sql.Expression) sql.Table {
+	nt := *r
+	nt.filters = filters
+	return &nt
 }
 
-func (t *filesTable) TransformUp(f sql.TransformNodeFunc) (sql.Node, error) {
-	return f(t)
+func (r *filesTable) WithProjection(colNames []string) sql.Table {
+	return r
```

Comment:
doesn't accept this table pushdowns of projections?

Reply:
same answer